src/worker/plugins/run.ts

```diff
@@ -57,6 +58,10 @@ export async function runProcessEvent(server: PluginsServer, event: PluginEvent)
 
             try {
                 returnedEvent = (await processEvent(returnedEvent)) || null
+                if (returnedEvent.team_id != team_id) {
+                    returnedEvent = null // don't try to ingest events with modified teamIDs
+                    throw new Error('Changing team IDs is an illegal operation')
```

Comment:
Nit: Can we devise an error message that will be more informative to a user of the plugin? Since plugin logs errors will now pop up on the UI so it'd be nice to do something like "Illegal operation: plugin tried to change team ID"